parlai/tasks/airdialogue/agents.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import FixedDialogTeacher
+from .build import build
+import os
+import json
+
+
+def _path(opt):
+    # build the data if it does not exist
+    build(opt)
+
+    # set up path to data (specific to each dataset)
+    jsons_path = os.path.join(
+        opt['datapath'], 'airdialogue', 'airdialogue_data', 'airdialogue'
+    )
+    return jsons_path
+
+
+class AirDialogueTeacher(FixedDialogTeacher):
+    """
+    AirDialogue Teacher.
+
+    This also contains other files (dev_kb.json, train_kb.json) with flight data about
+    return flights, price, connections, flight airlines, departure times, and other
+    flight information.
+    """
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+        jsons_path = _path(opt)
+        self.datatype = opt['datatype'].split(':')[0]
+        self.messages = []
+        self._setup_data(jsons_path)
```

Comment:
when we do batching, we have batchsize_num teachers. to avoid loading this data batchsize_num times, how about we share `self.messages` across instances? for example
```
if shared is not None:
     self.messages = shared['messages']
else:
     self.messages = self._setup_data(jsons_path)
```
you will also have to override the `share` function to share messages